### PR TITLE
Raise error when computed global is set or reset.

### LINKED ISSUE #7368: Don't allow setting a global if it was declared with a computable expression
Schema:

```
module default {
    global test := 'hello';
}
```

```
> set global test := 'world'; <--- this should error out, but it does not
OK

> select global test;
{'hello'}    <--- because this is confusing
```

### PR DESCRIPTION
**Pull request by dnwpark**
After `set global foo := false;`, attempting to run `set global lol := false;` produces the following error:
```
error: ConfigurationError: global 'foo' is computed from an expression and cannot be modified
  ┌─ <query>:1:12
  │
1 │ set global foo := false;
  │            ^^^ error
```

close #7368

### CODE REVIEW

File: edb/edgeql/compiler/config.py [hunk 1]
```diff
@@ -292,11 +292,20 @@ def _validate_config_object(
...
     glob = ctx.env.get_schema_object_and_track(
-        s_utils.ast_ref_to_name(expr.name), expr.name,
+        glob_name, expr.name,
         modaliases=ctx.modaliases, type=s_globals.Global)
     assert isinstance(glob, s_globals.Global)
 
+    if isinstance(expr, (qlast.ConfigSet, qlast.ConfigReset)):
+        if glob.get_expr(ctx.env.schema):
+            raise errors.ConfigurationError(
+                f"global '{glob_name}' is computed from an expression and "
```
Comment (msullivan) [inline edb/edgeql/compiler/config.py:304]: I'd probably just say "computed global 'whatever' may not be modified" or some such

**Review by msullivan — APPROVED** [review]
This seems good for `set global`. We want to also check what happens when the client specifies the values via the protocol

**Comment by dnwpark** [comment]
@msullivan I saw the issue mentioned that. With these changes, it seems that setting the global silently deletes it from the state.

**Comment by msullivan** [comment]
> @msullivan I saw the issue mentioned that. With these changes, it seems that setting the global silently deletes it from the state.

I'm not sure what you mean by that

**Review by msullivan — APPROVED** [review]

**Comment by dnwpark** [comment]
Also added a test to the python client here: https://github.com/edgedb/edgedb-python/pull/494

**Pull request merged**